### PR TITLE
Cleanup deprecated usage of gpu

### PR DESCRIPTION
**Pull request by Gkrumbach07**
closes: https://github.com/opendatahub-io/odh-dashboard/issues/1894

## Description

this pr does the following
- removes backend `/gpu` and all related utils, types, frontend services, frontend UI components, hooks that use that api endpoint
- moves model of accelerator profile out of openshift file and into odh file
- renames variables, components, params, hooks, utils so that
   - when talking about hardware accelerators (identifiers, detected counts) then you dont need to use profiles in the name
   - when talking about actual accelerator _profiles_ (the CRD), then I changed the name to reflect that. In some cases for looping functions i use the name `cr` (custom resource)
- renamed two modal texts to be clear it is talking about accelerator profiles and not accelerators (wording has been used from what Katie suggested)
- Did not touch manifests or annotation names

## How Has This Been Tested?

- all tests pass
- run through e2e flow of:
    -  creating, editing, removing accelerator profiles
    -  setting recommended accelerators for byon
    - making sure workbenches, runtimes, and notebook spawner all get the correct, toelrations, annotations, and resources from the selected accelerator

## Test Impact

all tests pass. had to edit delete test bc name of button changed

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

_no UI changes, just naming of code. any wording changes have already been verified by Katie_

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Can you actually remove gpuSetting from the DashboardConfig CRD -- I'm fairly certain this will just drop the value from the config since it's not longer used or needed. Please test though.

File: frontend/src/pages/projects/screens/spawner/spawnerUtils.ts [hunk 1]
```diff
@@ -161,9 +161,7 @@ export const isCompatibleWithAccelerator = (
     return false;
   }
 
-  return getCompatibleAcceleratorIdentifiers(obj).some(
-    (accelerator) => accelerator === acceleratorIdentifier,
-  );
+  return getCompatibleAcceleratorIdentifiers(obj).some((cr) => cr === acceleratorIdentifier);
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/spawner/spawnerUtils.ts:164]: Weird to call it `cr` 🤔 But variable names are not the end of the world...

**Review by andrewballantyne — APPROVED** [review]
Your description is a little off -- but otherwise this looks good to me.

File: frontend/src/pages/projects/screens/spawner/spawnerUtils.ts (hunk 1 above)
Comment (Gkrumbach07) [reply frontend/src/pages/projects/screens/spawner/spawnerUtils.ts:164]: oh yeah in that area it is weird i agree. should be something like _identifier_

**Pull request merged**